ako-gateway-api/lib/lib.go

```diff
@@ -151,7 +151,12 @@ func FindTargetPort(serviceName, ns string, svcPort int32, key string) intstr.In
 	}
 	return intstr.IntOrString{}
 }
-
+func IsListenerInvalid(gwStatus *gatewayv1.GatewayStatus, listenerIndex int) bool {
+	if len(gwStatus.Listeners) > int(listenerIndex) && gwStatus.Listeners[listenerIndex].Conditions[0].Type == string(gatewayv1.ListenerConditionAccepted) && gwStatus.Listeners[listenerIndex].Conditions[0].Status == "False" {
```

Comment:
listenerIndex is already of type int, we dont need to type convert it to int again.
can conditions[0] create index out of range error?

Reply:
Will take care of your comment in next.